### DISCUSSION TITLE
Enable use of item frame selectors with /accesstrust

### DISCUSSION THREAD

**Discussion by citizenengineer** [GriefPrevention/GriefPrevention#228] [Ideas]
### What behaviour is expected:

I'm trying to implement some red stone functions that are partially controlled using an item frame selector mechanism. This is meant to be publicly accessible in an otherwise claimed area. I was hoping to create a subdivision for the item frame block to permit the general public sufficient permission to rotate the object in the item frame, but not remove it or otherwise modify the blocks in the subdivision.

### What behaviour is observed:

Unfortunately, the only permission setting that allowed people to rotate the object in the item frame was a full /trust. This would also give them permission to modify the blocks in the subdivision, however.

### Steps/models to reproduce:

This video displays an example of the mechanism in question. Ideally, only the front block containing the item frame would be contained in the subdivision.

https://www.youtube.com/watch?v=T34SXQxebTE

### Spigot/Craftbukkit/Paper and GriefPrevention version:

AFAIK, this is applicable to all versions of this plugin.

The ideal solution would seem to be that item frames should be included in the /accesstrust permission - you should be able to operate the selector object (in the item frame), but not remove it. So I guess this counts as an enhancement. Basically, we want to treat selectors like switches and buttons.

**Comment by RoboMWM** [comment]
See discussion in #31

**Comment by rjohnson98126** [comment]
this functionality is necessary.  the use of redstone multiposition switches is an awesome feature of redstone builds and quests and puzzles, and must always used in public areas that otherwise need to be protected from damage.     

I read the complaints from some users in the prior discussion, because they were concerned about some people's decorations being tampered with (rotated)

rotating (but not breaking or taking contents from) item frames could be allowed in /containertrust.   this would prevent MOST builds from having random "vandalizing" occurred, as /containertrust is a pretty high level of trust, and even in the rare instance where a some admin area gives generic  `/containertrust public` for public resources depots AND has pretty decorations they dont want nudged, they would be merely _inconvenienced_ by having to subdivide a claim to allow appropriate permissions on their public containers.  And really, they wouldnt need to be inconvenienced at all, becasue this could be a configuration setting that is by default not enabled. 

Conversely, for people who build clever redstone machines with multiposition switches and/or npc/quest/puzzle selectors, they don't even _have_ an "inconvenient" option, their only option is to literally grant full /trust to public on those switches and just _hope_ no one breaks it.
